src/community-toolbox.js

```diff
@@ -39,6 +39,136 @@ CommunityToolbox = function CommunityToolbox(org, repo) {
        .then(callback);
   }
 
+  function fetchRepoContributors(org, repo) {
+
+    var arr = [];
+    var getPages = api.Repositories
+           .getRepoContributors(org, repo, {method: "HEAD", qs: { sort: 'pushed', direction: 'desc', per_page: 100 } })
+           .then(function(contribData) {
+             var headers = contribData;
+             if (headers.hasOwnProperty("link")) {
+                 var parsed = parse(headers['link']);
+                 totalPages = parseInt(parsed.last.page);
+             } else {
+                 totalPages = 1;
+             }
+             return totalPages;
+           });
+
+    // get data given the page number
+    function getData(curPage){
+      api.Repositories
+            .getRepoContributors(org, repo, { method:"GET", qs: { sort: 'pushed', direction: 'desc', per_page: 100, page:curPage } })
+            .then(function(contributors) {
+                if (contributors != null || contributors.length > 0) {
+                  contributors.map((contributor, i) => {
+                    arr.push(contributor);
+                  })
+                }
+            });
+    }
+
+    getPages.then(function(totalPages){
+      for(var i = 1; i <= totalPages; i++) {
+        getData(i);
+      }
+    });
+  console.log("\narr before return in fetchRepoContributors = ", arr, '\n')
+  return arr;
+}
+
+
+
+  // Fetches all the publiclab repos
+  function getAllRepos(org) {
+    let repos = [];
+    fetch(`https://api.github.com/orgs/${org}/repos`, {'headers': { "Content-Type": "application/x-www-form-urlencoded", "User-Agent": "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/68.0.3440.106 Safari/537.36" }})
+    .then((response) => {
+      if (response.status == "200") {
+        return response.json();
+      }
+    })
+    .then((res) => {
+      res.map((item, i) => {
+        repos[i] = item.name;
+      });
+      // Stores all public lab repos to localstorage (only first 30 repos yet)
+      localStorage.setItem('repos', JSON.stringify(repos));
+    })
+  }
+
+  // Fetches all the UNIQUE contributors from all the publiclab repos
+  // and stores them in localstorage in the form of array
+  function getAllContributorsInStorage(org) {
+    let contributorSet = new Set([]);
+    let myArr = [];
+    let repos = JSON.parse(localStorage.getItem('repos'));
+    console.log("In getAllContributorsInStorage,repos= ", repos, '\n')
+    var testRepos = repos.splice(0,2);
+    testRepos.map((repo, i) => {
+      var repoContributors = fetchRepoContributors(org, repo);
+      console.log("total contributors for repo ", i, "are : ", repoContributors ,repoContributors.length, '\n')
```

Comment:
Here, `repoContributors.length` is 0 while `repoContributors` has objects in it...What is the reason for this behaviour??